### DISCUSSION TITLE
revealjs: option to show text in the output slide when `#| output-location: slide` is selected

### DISCUSSION THREAD

**Discussion by Lulliter** [quarto-dev/quarto-cli#9446] [Q&A]
### Description

(Related to  [issues 4213](https://github.com/quarto-dev/quarto-cli/issues/4213))

In revealjs presentations, when `#| output-location: slide` is selected, it would be great to have **_also_** the option of showing some comments/annotations in the slide where the output is rendered (e.g. a plot). Below is a toy presentation to clarify:

````qmd
---
title: ""
format: revealjs
---

## Plot comment (falls in the wrong slide)

**Code comment**: Here you can see the code (OK!)  
```{r}
#| echo: true
#| eval: true
#| output-location: slide

library(dplyr)
library(ggplot2)

starwars %>% 
  filter(homeworld == c("Alderaan","Tatooine" )) %>% 
  ggplot( aes(x = height,fill = homeworld))+
  geom_density(alpha = 0.5)  
```

**Plot comment**:  "Here you can see how the distribution of variable `height` is not normal"  
````

**Comment by mcanouil** [comment]
Why not write your output in a code cell and use the option?

**Reply by Lulliter** [reply]
not sure what this means ...

**Reply by mcanouil** [reply]
Output your comment with the plot.

**Reply by Lulliter** [reply]
Oh maybe you mean like `# comment`? I guess that's an option, but with formatting inconsistency

**Reply by mcanouil** [reply]
````qmd
```{r}
#| output: asis
#| output-location: slide
plot(1)
cat("\nA comment\n")
```
````

This a quick and dirty example written on a phone.

**Reply by cscheid** [reply]
High-quality plot annotations will usually require library-specific code. I would refer to the [annotations](https://ggplot2.tidyverse.org/reference/index.html#annotations) part of ggplot2's documentation for more.

**Reply by Lulliter** [reply]
that makes sense! thanks

**Reply by mcanouil** [reply]
Did not try but you could also emit a regular cross-referenceable figure with a caption. (It should work and it feels better than having "random" text on the same slide)

**Reply by Lulliter** [reply]
For my purposes creating cross-referenceable figures is too much work (this is a class demonstrating several ggplot graphs in a row), but I think `patchwork` can be a good compromise for now.
 
````qmd
---
title: ""
format: revealjs
---

## This is what I wanted but can't get rid of `cat(...)`
```{r}
#| echo: true
#| eval: true
#| output-location: slide

library(dplyr)
library(ggplot2)

starwars %>% 
  filter(homeworld == c("Alderaan","Tatooine" )) %>% 
  ggplot( aes(x = height,fill = homeworld))+
  geom_density(alpha = 0.5)  

cat("\nHere you see how the distribution is not normal\n")
```

## Maybe `patchwork` is a good workaround for now
```{r}
#| echo: true
#| eval: true
#| output-location: slide

library(dplyr)
library(ggplot2)
library(patchwork)

plot <- starwars %>% 
  filter(homeworld == c("Alderaan","Tatooine" )) %>% 
  ggplot( aes(x = height,fill = homeworld))+
  geom_density(alpha = 0.5)  

plot + plot_annotation(title = "Here you see how the distribution is not normal" )
```   
````

**Reply by mcanouil** [reply]
Too much work to add `#| fig-cap: "Your caption/comment"`?

**Reply by Lulliter** [reply]
right! that works too
